static/scripts/orbit-camera.js

```diff
@@ -1,3 +1,57 @@
+
+var transitionTime = 0.4;
+
+var SmoothedValue = function (value) {
+    this._timer = 0;
+    this._value = value;
+    this._startValue = value;
+    this._targetValue = value;
+}
+
+Object.defineProperties(SmoothedValue.prototype, {
+    value: {
+        get: function () {
+            return this._value;
+        },
+
+        set: function (value) {
+            this._timer = 0;
+            this._startValue = this._value;
+            this._targetValue = value;
+        }
+    }
+});
+
+Object.assign(SmoothedValue.prototype, {
+    snapTo: function (value) {
+        this._value = value;
+    },
+
+    // snap to target
+    snap: function () {
+        this._value = this._targetValue;
+        this._timer = transitionTime;
+    },
+
+    distance: function () {
+        return this._targetValue - this._value;
+    },
+
+    update: function (dt) {
+        if (this._timer < transitionTime) {
+            this._timer = Math.min(this._timer + dt, transitionTime);
+            var n = this._timer / transitionTime;
+            // var t = Math.sin(n * Math.PI / 2.0);        // sinosidal
+            // var t = n * (2 - n);                        // quadratic
+            // var t = 1 - --n * n * n * n;                // quartic
+            var t = --n * n * n * n * n + 1;            // quintic
+            this._value = pc.math.lerp(this._startValue, this._targetValue, t);
+        } else {
+            this._value = this._targetValue;
+        }
+    }
+});
```

Comment:
Are we OK with this script diverging from the orbit-camera.js? Wouldn't we want to maintain/improve a 'single' version of it?

Reply:
we probably can't radically change the packaged version of orbit camera and I'd like to convert this implementation to es6 modules for the viewer.